File: MyceliumCore/src/main/java/net/jibini/mycelium/Mycelium.java
package net.jibini.mycelium;

import net.jibini.mycelium.thread.NamedThread;

public final class Mycelium
{
	public static final MyceliumSpore SPORE = new MyceliumSpore();
	
	public static void main(String[] args)
	{
		new NamedThread()
				.withName("MainThread")
				.withRunnable(() -> SPORE.start())
				.start();
	}
}

File: MyceliumCore/src/main/kotlin/net/jibini/mycelium/Mycelium.kt
package net.jibini.mycelium

import net.jibini.mycelium.document.DocumentMessage
import net.jibini.mycelium.transfer.Patch
import net.jibini.mycelium.transfer.Router

object Mycelium
{
    @JvmStatic
    val localRouter = Router.create<DocumentMessage>()

    init
    {
        localRouter.embedReturns = true
        localRouter.routeBy = "target"
    }

    @JvmStatic
    fun createSpore(address : String) : MyceliumSpore
    {
        val patch = Patch.createAsync<DocumentMessage>()
        localRouter.attach(address, patch.uplink)

        return MyceliumSpore.create(patch)
    }
}